Sources/CombineX/Internal/Cache.swift

```diff
@@ -0,0 +1,74 @@
+import CXUtility
+
+class Cache<Key: Hashable, Value> {
+    
+    private var storage: [Key: Value] = [:]
+    
+    private var lock = Lock()
+    
+    func value(for key: Key) -> Value? {
+        return storage[key]
+    }
+    
+    func value(for key: Key, make: () throws -> Value) rethrows -> Value {
+        if let value = storage[key] {
+            return value
+        }
+        lock.lock()
+        defer { lock.unlock() }
+        if let value = storage[key] {
+            // avoid double write
+            return value
+        }
+        let value = try make()
+        storage[key] = value
+        return value
+    }
+}
```

Comment:
No `remove` method? I know the usage scenarios of this class, but its name(cache) indicates that it is a general-purpose cache.

Reply:
It is general-purpose as long as you don't modify cached value. The same rule applied to `TypeInfo` cache and `ObservableObjectPublisher` cache (with strong and weak variant). so what's the appropriate name?

- ImmutableCache
- WriteOnceCache
- OneOffCache